src/components/ButtonPanel.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import Button from './Button';
+
+const genKey = () => Math.floor(Date.now() * Math.random());
+
+const renderGroup = group => (
+  group.map(item => <Button name={item} key={genKey()} />)
+);
+
+const ButtonPanel = () => {
+  const groups = [
```

Comment:
* [ ] It includes all the following buttons divided in the groups below: ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/buttonPanel/buttons.png)).
Group 1: AC, +/-, %, ÷
Group 2: 7, 8, 9, X
Group 3: 4, 5, 6, -
Group 4: 1, 2, 3, +
Group 5: 0, ., =
let us change the array to hash table with key as group name and value as the corresponding value as we have above.

Reply:
The purpose of the array is only to hold the strings that are going to be iterated over to create the buttons. The group names for this structure aren't referenced at any point. I'm implementing the change, but the key names are redundant.